### DISCUSSION TITLE
Vocês sentem que mesmo estudando sempre, não se sentem preparados?

### DISCUSSION THREAD

**Discussion by renedocket** [frontendbr/forum#1884] [Discussão]
Olá pessoal, tudo bem? 

Bom venho aqui falar de algo que estou vivenciando em relação a nossa área, estou a pouco tempo na área, fiz 2 anos recentemente na área e desde do meu inicio eu evoluí muito. Mesmo sabendo que tem um monte de coisas pra aprender eu tento sempre focar no básico, porém vejo que não é só de linguagens de programação que se vive, temos muitos conceitos de arquitetura de software, de designer patterns e até mesmo as nossas soft skills que temos que focar.

Eu sinto que existem muitas coisas pra focarmos e mesmo focando somente uma por vez, tenho dificuldade se o caminho está fazendo sentindo sabe? 

Eu gostaria de saber a opinião de vocês sobre esse assunto, se sentem o mesmo, o que fazem, pra onde estão indo?

**Comment by gabriel-hahn** [comment]
Opa @renedocket, cara acho que isso é completamente normal na nossa área em que tudo muda/evolui muito rápido e é praticamente impossível saber tudo. Esse teu sentimento acho que qualquer pessoal vai sentir ou já sentiu várias vezes durante a carreira. Nunca vamos nos sentir _**prontos**_ para algo, mesmo possuindo bastante conhecimento naquilo. Tu mesmo já disse que evoluiu muito, garanto que tem muitas coisas que tu não fazia ideia de como construir a 2 anos, porém agora tu já sabe e tens certo domínio daquilo.

Algo interessante é ir adquirindo experiências e ir focando no que mais gosta. Eu já trabalhei como QA, somente com apps desktop, depois como full-stack e hoje estou focando em Front-end que é o que eu mais gosto e curto trabalhar. Hoje, tento me aprofundar nas linguagens e arquitetura em volta do Front-end, mas não necessariamente deixo de estudar um pouco de tudo e 'me virar' caso tenha necessidade, mas não saberia/ou demoraria muito tempo para resolver problemas mais complicados de Back-end por exemplo. Sobre soft skills, acho que depende mais da tua experiência na área e personalidade mesmo, algo não tão palpável para te responder aqui hehe

Uma dica seria tu realmente encontrar o que tu mais gosta e focar nisso, por exemplo, ser dev mobile e ir construindo projetos pessoais, pegar um trampo nessa área, dar uma olhada em _roadmaps_ e no que o mercado está exigindo e ir adquirindo experiência. Não tem um caminho curto para ser _expert_ em algo e esse teu sentimento é super normal e comum. Abraço!

**Comment by renedocket** [comment]
@gabriel-hahn poxa, muito obrigado! Visão muito esclarecedora, fico até mais tranquilo em saber que esse sentimento é normal entre nós. Espero que outras pessoas também se identificado! <3 Muito obrigado, Abraço!

**Comment by Rafaelzf** [comment]
Fala @renedocket não se preocupe isso é normal. Vou além e digo que isso é saudável, pois mostra que sua percepção sobre a realidade é apurada, pois não há como sabermos tudo, é MUITA COISA. Entretanto, saber que não sabe tudo, ou saber pouco comparado ao total de informações existentes é diferente de não saber nada ou não saber nem o suficiente.

**Comment by mathiasfc** [comment]
> Fala @renedocket não se preocupe isso é normal. Vou além e digo que isso é saudável, pois mostra que sua percepção sobre a realidade é apurada, pois não há como sabermos tudo, é MUITA COISA. Entretanto, saber que não sabe tudo, ou saber pouco comparado ao total de informações existentes é diferente de não saber nada ou não saber nem o suficiente.

Fala @Rafaelzf blz? Curti a tua maneira de encarar esse sentimento como algo saudável, muito massa mesmo! Direto fico preocupado por não saber sobre determinado assunto mas essa preocupação é justamente uma das coisas que me motivam a ir atrás e me aprofundar :clap:

**Comment by joaroque** [comment]
Muito normal entre os devs

**Comment by dionarodrigues** [comment]
Às vezes eu tenho a sensação que podemos passar a vida toda estudando na área de dev e mesmo assim ter a sensação de não saber o suficiente ou não ser bom o suficiente, pois sempre há um montão de coisas para aparender, hehehe. Bom, já dizia ~Sócrates~ Clarice Lispector: "só sei que nada sei". 

Penso que assim como o pessoal disse, o importante é definir uma tecnologia que você tem vontade de aprender mais a fundo -e que também seja valorizada no mercado de trabalho - e seguir adiante. Em pararelo, tentar acompanhar as novidades da área só pra ficar a par do que está acontecendo; fazer um teste aqui, outro acolá; de repente, criar um blog pra estimular a busca por novos conhecimentos; deixar o GitHub mais ativo, inserindo seus experimentos lá... 

Aqui tem dois [artigos bem interessante sobre esse tema](https://dev.to/wilsonrc).

E recentemente eu criei um artigo onde listo [alguns dos web designers/developers que eu costumo seguir no Twitter](https://www.diogorodrigues.dev/pt/blog/web-designers-e-desenvolvedores-front-end-para-seguir-no-twitter).  ✨👍 Isso me ajuda a ficar atualizado, mesmo não me aprofundando muito no que está rolando, enquanto me dedico à algumas techs específicas. Fica a dica ;)

**Comment by xenodesire** [comment]
> Olá pessoal, tudo bem?
> 
> Bom venho aqui falar de algo que estou vivenciando em relação a nossa área, estou a pouco tempo na área, fiz 2 anos recentemente na área e desde do meu inicio eu evoluí muito. Mesmo sabendo que tem um monte de coisas pra aprender eu tento sempre focar no básico, porém vejo que não é só de linguagens de programação que se vive, temos muitos conceitos de arquitetura de software, de designer patterns e até mesmo as nossas soft skills que temos que focar.
> 
> Eu sinto que existem muitas coisas pra focarmos e mesmo focando somente uma por vez, tenho dificuldade se o caminho está fazendo sentindo sabe?
> 
> Eu gostaria de saber a opinião de vocês sobre esse assunto, se sentem o mesmo, o que fazem, pra onde estão indo?

Acho que todo mundo ja passou por isso, o que importa no final é como você lida com isso.
